File: ruby/word-count/word_count.rb
class Phrase
  def initialize(string)
    @words = string.split(/[ ,\n]/)
  end

  def word_count
    @words.map {|w| clean(w)}
          .reject(&:empty?)
          .each_with_object(Hash.new(0)) do |word, counts|
            counts[word] += 1
          end
  end

  private
  def clean(word)
    word.downcase.gsub(/\W+$/,'')
  end
end

File: javascript/word-count/word-count.js
var words = function(phrase) {
  return phrase.split(/\s/)
    .reduce(function(hist, word) {
      if (word.trim().length == 0) return hist

      hist[word] = hist[word] || 0
      hist[word] += 1
      return hist
    }, Object.create(null))
}
module.exports = words;
